### ISSUE TITLE
G26 — Cross-Domain Regression Matrix

### ISSUE THREAD

**Issue by samscarrow** [git-scarrow/fast-associative-memory#37]
## Gauntlet 26 — Cross-Domain Regression Matrix

**Phase:** Phase 4 – Exploration
**Kill Condition:** Any single-domain adapter (Birds, Cars, Aircraft, Flowers) drops accuracy > 5pp on CIFAR-100 or ImageNet-R relative to no-adapter baseline. Same protocol as G19.

**G19 Reference:** Dogs adapter on CIFAR-100 dropped +2.03pp, on ImageNet-R dropped −1.15pp (adapter actually improved). Kill threshold: 5pp.

---

### Objective

Create a script `benchmarks/g26_cross_domain_matrix.py` that extends the G19 cross-domain regression protocol to all 4 new domain-specific adapters from G22–G25.

The script must:

1. **Load all 4 domain-specific adapter weight files:**
   - `adapter_cub200_birds.pt` (G22)
   - `adapter_stanford_cars.pt` (G23)
   - `adapter_fgvc_aircraft.pt` (G24)
   - `adapter_flowers102.pt` (G25)
   Each adapter is `MetricAdapter(input_dim=1024, output_dim=256, hidden_dim=512)`.

2. **Load pre-cached CIFAR-100 and ImageNet-R feature caches:**
   - CIFAR-100: `--cifar-cache ./feature_cache_vitb14` (existing, contains `*train*.pt` and `*test*.pt`)
   - ImageNet-R: `--imagenet-r-cache ./feature_cache_imagenet_r_vitl14` (existing)
   
   These are the same feature caches used by `benchmarks/gauntlet_19_cross_domain.py`.

3. **Run the G19 protocol for each adapter × each dataset (4 × 2 = 8 evaluations):**
   - For each dataset: evaluate FAM without adapter (baseline), then evaluate FAM with each adapter
   - Compute accuracy drop = baseline − adapted (positive = degradation)
   - Check kill condition: drop > 5.0 pp

4. **Output a full regression matrix:**

```
G26 — Cross-Domain Regression Matrix
=========================================================================
  Dataset       Baseline    Birds     Cars      Aircraft  Flowers
  -------------------------------------------------------------------------
  CIFAR-100     XX.XX%      XX.XX%    XX.XX%    XX.XX%    XX.XX%
    drop (pp)               +X.XX     +X.XX     +X.XX     +X.XX
  ImageNet-R    XX.XX%      XX.XX%    XX.XX%    XX.XX%    XX.XX%
    drop (pp)               +X.XX     +X.XX     +X.XX     +X.XX
=========================================================================

Kill condition (> 5.0 pp drop): [list any failures, or "NONE — all adapters pass"]
→ PASS / KILL
```

### Implementation Details

- Reuse `eval_fam_accuracy()` and `load_features()` from `benchmarks/gauntlet_19_cross_domain.py` — either import directly or copy the functions.
- Each adapter is loaded with `MetricAdapter(input_dim=1024, output_dim=256, hidden_dim=512)` and `adapter.load_state_dict(torch.load(...))`.
- The baseline (no adapter) only needs to be evaluated once per dataset.
- If any adapter file is missing, print a warning and skip that adapter (don't crash).
- Exit code 0 if all pass, 1 if any kill condition met.

### CLI Interface

```
python benchmarks/g26_cross_domain_matrix.py \
  --cifar-cache ./feature_cache_vitb14 \
  --imagenet-r-cache ./feature_cache_imagenet_r_vitl14 \
  --adapter-birds adapter_cub200_birds.pt \
  --adapter-cars adapter_stanford_cars.pt \
  --adapter-aircraft adapter_fgvc_aircraft.pt \
  --adapter-flowers adapter_flowers102.pt \
  [--threshold 5.0] [--seed 42]
```

### Files to Create
- `benchmarks/g26_cross_domain_matrix.py`

### Files to Reference (do not modify)
- `benchmarks/gauntlet_19_cross_domain.py` — G19 protocol (primary reference)
- `adapter.py` — `MetricAdapter` class
- `fast_associative_memory.py` — `FastAssociativeMemory` class
